pkg/transport/listener.go

```diff
@@ -174,6 +178,23 @@ func (info TLSInfo) baseConfig() (*tls.Config, error) {
 		MinVersion:   tls.VersionTLS12,
 		ServerName:   info.ServerName,
 	}
+
+	if info.RequireCN != "" {
+		cfg.VerifyPeerCertificate = func(rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error {
+			if len(verifiedChains) != 0 {
+				chains := verifiedChains[0]
```

Comment:
Q. why `verifiedChains[0]`? Do we ever get more than 1 `[]*x509.Certificate`?

Reply:
I believe the first certificate is the leaf certificate.

Reply:
@mitake Let's document that on the code, then.
Thanks!

Reply:
`verifiedChains` is documented as `any verified chains that normal processing found`. Wouldn't it be safest to verify them all (`for _, chains := range verifiedChains { ...`), just in-case there is more than one?  I can't think of any down side.

Reply:
Sorry, you're right. The first cert of an individual chain is the leaf. Checking each chain sounds correct.

Reply:
@gyuho @ericchiang @jpbetz ok, I'll change the mechanism of extracting CN later. Thanks for your comments.